backend/src/main/java/com/htecgroup/SkyNest/service/impl/EmailServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.htecgroup.SkyNest.service.impl;
+
+import com.htecgroup.SkyNest.service.EmailService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+@Service
+public class EmailServiceImpl implements EmailService {
+  @Autowired private JavaMailSender javaMailSender;
+  private final Logger LOGGER = LoggerFactory.getLogger(EmailServiceImpl.class);
+
+  @Override
+  @Async
+  public void send(String to, String email) {
+    try {
+      MimeMessage mimeMessage = javaMailSender.createMimeMessage();
+      MimeMessageHelper helper = new MimeMessageHelper(mimeMessage, "utf-8");
+      helper.setText(email, true);
```

Comment:
I would try to make this helper a reusable method, cause maybe there would be different mails that needs to be sent and the code is pretty much the same, except to, email and subject

Reply:
yes, you can even create an object that will contain every parameter needed for sending different mail. Fields such as: to, from, text, html, subject etc etc